core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -986,7 +986,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     union(Seq(first) ++ rest)
 
   /** Get an RDD that has no partitions or elements. */
-  def emptyRDD[T: ClassTag] = new EmptyRDD[T](this)
+  def emptyRDD[T: ClassTag]: EmptyRDD[T] = new EmptyRDD[T](this)
```

Comment:
This is written up a bit more in https://issues.apache.org/jira/browse/SPARK-2331 which should be reopened for the new "2+" bucket in JIRA. This should be fixed when binary compatibility can be broken. Yes, big +1 to tightening up types like this.